File: android/app/src/main/java/de/repeatuntil/yata/domain/user/UserRepository.kt
package de.repeatuntil.yata.domain.user

interface UserRepository {

    suspend fun findActiveUser(): Result<User?>

    suspend fun getAllUsers(): Result<List<User>>

    suspend fun updateUsers(users: List<User>): Result<Unit>

    suspend fun addUser(user: User): Result<Unit>

    suspend fun updateUser(user: User): Result<Unit>
}

File: ios/YATA/Domain/User/UserRepository.swift
//
//  UserRepository.swift
//  YATA
//
//  Created by Alpár Szotyori on 26.05.24.
//

import Foundation

protocol UserRepository {

    func findActiveUser() async -> Result<User?, Error>

    func getAllUsers() async -> Result<[User], Error>

    func updateUsers(_ users: [User]) async -> Result<Void, Error>

    func addUser(_ user: User) async -> Result<Void, Error>

    func updateUser(_ user: User) async -> Result<Void, Error>

}
